src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/ServiceControllerTests.cs

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.ServiceProcess;
+using System.Threading;
+using Xunit;
+using Microsoft.Win32;
+
+namespace System.ServiceProcessServiceController.Tests
+{
+    internal sealed class ServiceProvider
+    {
+        public readonly string TestMachineName;
+        public readonly TimeSpan ControlTimeout;
+        public readonly string TestServiceName;
+        public readonly string TestServiceDisplayName;
+        public readonly string DependentTestServiceNamePrefix;
+        public readonly string DependentTestServiceDisplayNamePrefix;
+        public readonly string TestServiceRegistryKey;
+
+        public ServiceProvider()
+        {
+            TestMachineName = ".";
+            ControlTimeout = TimeSpan.FromSeconds(3);
+            TestServiceName = Guid.NewGuid().ToString();
+            TestServiceDisplayName = "Test Service " + TestServiceName;
+            DependentTestServiceNamePrefix = TestServiceName + ".Dependent";
+            DependentTestServiceDisplayNamePrefix = TestServiceDisplayName + ".Dependent";
+            TestServiceRegistryKey = @"HKEY_USERS\.DEFAULT\ToF\ServiceController\" + TestServiceName;
+
+            // Create the service
+            CreateTestServices();
+        }
+
+        private void CreateTestServices()
+        {
+            // Create the test service and its dependent services. Then, start the test service.
+            // All control tests assume that the test service is running when they are executed.
+            // So all tests should make sure to restart the service if they stop, pause, or shut
+            // it down.
+            RunServiceExecutable("create");
+        }
+
+        public void DeleteTestServices()
+        {
+            RunServiceExecutable("delete");
+            RegistryKey users = Registry.Users;
+            if (users.OpenSubKey(".DEFAULT\\ToF") != null)
```

Comment:
I think it would be cleaner to put this in its own key: .DEFAULT\ServiceControllerTests
instead of putting it in a subkey under ToF, which means nothing in the open-source repo.  Also, if other tests happen to use the ToF key, we'd be blowing away those subkeys when we finish these tests (I don't think that's a big deal though).